### ISSUE TITLE
Script Events which contain EntityID data pin cause an Assert when launching Script Canvas

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#7620]
**Describe the bug**
Launching Script Canvas after creating a Script Event which contain EntityID Parameter (or a Return / Address type) results in the following Assert being thrown in the Console:

**File**: `D:/Git/o3de/Code/Framework/AzCore/AzCore/Script/ScriptContext.cpp`
**Line**: `3430`
**Condition**: 
```
(7840) '__cdecl AZ::LuaScriptCaller::LuaScriptCaller(class AZ::BehaviorContext *,class AZ::BehaviorMethod *)'
(System) - The argument type: ParameterName for method: EntityIDMethod is not serialized and/or reflected for scripting.
Make sure ParameterName is added to the SerializeContext and reflected to the BehaviorContext
For example, verify these two exist and are being called in a Reflect function:
serializeContext->Class<ParameterName>();
behaviorContext->Class<ParameterName>();
EntityIDMethod will not be available for scripting unless these requirements are met.
```

Please refer to the attached video and log files for more details.

**Steps to reproduce**
1. Open any Level.
2. Open Asset Editor and create a new Script Event with a Method that contains an EntityID type Parameter.
3. Save the Script Event.
4. Open Script Canvas.

**Expected behavior**
No Asserts are displayed in the Console.

**Actual behavior**
An Assert is displayed in the Console.

**Assets required**
N/A

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/153878396-2583cbe5-82fe-480f-b97a-48266ba84b32.mp4

**Found in Branch**
Development (a3ebe61)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

**Attachments**
[Assert.txt](https://github.com/o3de/o3de/files/8061300/Assert.txt)
[Editor.log](https://github.com/o3de/o3de/files/8061301/Editor.log)

**Comment by lawsonamzn** [comment]
So I did an investigation of why this is happening and this is the result.

In O3DE, there's a concept of a Behavior Context.  This is a structure (ie, an unordered map, really) that you can give the typeid of a class, and it will tell you about that class from a scripting point of view - what methods it has, what member variables it has, etc.

When the app starts up, it passes the Behavior Context pointer around, and various classes and structures get a chance to declare themselves to this context, telling it what they are, what members they have, etc.  All of this is stored in a map of [typeid] -> [info about it] in the behavior context.

When the Script System COmponent (ie, the lua component) needs to convert from a blind type in LUA, into a type in C++, and its not one of the primitive types, it looks in the behavior context map of [typeid] -> information to find that type, by type ID.

Whats happening here is that its going to go look at this map and tries to look up AZ::EntityId, by its typeid.  But the map has got no such type in it.

Why?

Because there are actually multiple behavior contexts and the wrong one is bound at that moment to the LUA Script Context.  The Material scripting system seems to create its own behavior context, and pushes its own types into it.  Whenever a Material Scripting System asset loads, it switches the global singleton LUA Script Context to point at its own behavior context pointer instead of the AZ::Core application one.  And it does not switch it back.

This means that any time after a Material Scripting type asset loads (even on another thread), the global singleton LUA Script Context will then be pointing at the wrong Behavior Context via the BindTo() function.   From that point on, any LUA operations involving non Material Scripting components will fail.

**Comment by LB-TomaszSerwanski** [comment]
Fix verified in Development (60adc09).
